assignment-client/src/audio/AudioMixer.cpp

```diff
@@ -535,36 +544,45 @@ void AudioMixer::sendAudioEnvironmentPacket(SharedNodePointer node) {
 }
 
 void AudioMixer::readPendingDatagram(const QByteArray& receivedPacket, const HifiSockAddr& senderSockAddr) {
-    auto nodeList = DependencyManager::get<NodeList>();
-
     if (nodeList->packetVersionAndHashMatch(receivedPacket)) {
-        // pull any new audio data from nodes off of the network stack
-        PacketType::Value mixerPacketType = packetTypeForPacket(receivedPacket);
-        if (mixerPacketType == PacketType::MicrophoneAudioNoEcho
-            || mixerPacketType == PacketType::MicrophoneAudioWithEcho
-            || mixerPacketType == PacketType::InjectAudio
-            || mixerPacketType == PacketType::SilentAudioFrame
-            || mixerPacketType == PacketType::AudioStreamStats) {
-
-            nodeList->findNodeAndUpdateWithDataFromPacket(receivedPacket);
-        } else if (mixerPacketType == PacketType::MuteEnvironment) {
-            SharedNodePointer sendingNode = nodeList->sendingNodeForPacket(receivedPacket);
-            if (sendingNode->getCanAdjustLocks()) {
-                auto packet = NLPacket::create(PacketType::MuteEnvironment);
-                // Copy payload
-                packet->write(receivedPacket.mid(numBytesForPacketHeader(receivedPacket)));
-
-                nodeList->eachNode([&](const SharedNodePointer& node){
-                    if (node->getType() == NodeType::Agent && node->getActiveSocket() &&
-                        node->getLinkedData() && node != sendingNode) {
-                        nodeList->sendPacket(std::move(packet), node);
-                    }
-                });
+        nodeList->processNodeData(senderSockAddr, receivedPacket);
+    }
+}
```

Comment:
Yeah - after this morning's convo, I think it's safe to not call `nodeList->processNodeData` and replace that functionality in PacketReceiver